### PR TITLE
Rdm 5398

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
RDM-5398: Document Upload fields within mandatory complex and collection fields are not validated.

    [RDM-5398](https://tools.hmcts.net/jira/browse/RDM-5398).

### CODE REVIEW

File: src/shared/components/palette/document/write-document-field.component.ts [hunk 1]
```diff
@@ -198,9 +210,17 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
     if (this.uploadedDocument) {
-      return this.uploadedDocument.get('document_filename').value;
+      return this.uploadedDocument.get(WriteDocumentFieldDefinitios.DOCUMENT_FILENAME).value;
     } else {
       return undefined;
     }
   }
 }
+
+class WriteDocumentFieldDefinitios {
```
Comment (tzarsmango) [inline src/shared/components/palette/document/write-document-field.component.ts]: typo here missing `n` towards the end --> WriteDocumentFieldDefinitions

File: src/shared/components/palette/document/write-document-field.component.ts [hunk 2]
```diff
@@ -198,9 +210,17 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
+  static readonly DOCUMENT_URL = 'document_url';
+  static readonly DOCUMENT_BINARY_URL = 'document_binary_url';
+  static readonly DOCUMENT_FILENAME = 'document_filename';
+  static readonly UPLOAD_ERROR = 'File required';
+  static readonly UPLOAD_ERROR_NO_AVAILABLE = 'Document upload facility is not available at the moment';
```
Comment (tzarsmango) [inline src/shared/components/palette/document/write-document-field.component.ts]: would it be better if we change to `UPLOAD_ERROR_NOT_AVAILABLE`

File: src/shared/components/palette/document/write-document-field.component.ts [hunk 3]
```diff
@@ -198,9 +210,17 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
       return undefined;
     }
   }
 }
+
+class WriteDocumentFieldDefinitios {
+  static readonly DOCUMENT_URL = 'document_url';
+  static readonly DOCUMENT_BINARY_URL = 'document_binary_url';
+  static readonly DOCUMENT_FILENAME = 'document_filename';
+  static readonly UPLOAD_ERROR = 'File required';
```
Comment (tzarsmango) [inline src/shared/components/palette/document/write-document-field.component.ts]: Would it be better if we change to `UPLOAD_ERROR_FILE_REQUIRED`

File: src/shared/components/palette/document/write-document-field.component.ts (hunk 1 above)
Comment (Thor-tech-of-metal) [reply src/shared/components/palette/document/write-document-field.component.ts]: DONE

File: src/shared/components/palette/document/write-document-field.component.ts (hunk 3 above)
Comment (Thor-tech-of-metal) [reply src/shared/components/palette/document/write-document-field.component.ts]: DONE :)

**Pull request merged**